src/python/pants/fs/fs_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+
+from pants.engine.console import Console
+from pants.engine.fs import (
+  Digest,
+  DirectoryToMaterialize,
+  FileContent,
+  InputFilesContent,
+  MaterializeDirectoriesResult,
+  MaterializeDirectoryResult,
+  Workspace,
+)
+from pants.engine.goal import Goal
+from pants.engine.rules import console_rule
+from pants.engine.selectors import Get
+from pants.util.contextutil import temporary_dir
+from pants_test.console_rule_test_base import ConsoleRuleTestBase
+from pants_test.test_base import TestBase
+
+
+class TestWorkspaceGoal(Goal):
+  name = 'test-workspace-goal'
+
+
+@console_rule
+def workspace_console_rule(console: Console, workspace: Workspace) -> TestWorkspaceGoal:
+  input_files_content = InputFilesContent((
+    FileContent(path='a.txt', content=b'hello', is_executable=False),
+  ))
+  digest = yield Get(Digest, InputFilesContent, input_files_content)
+  with temporary_dir() as tmp_dir:
+    output = workspace.materialize_directories((
+      DirectoryToMaterialize(path=tmp_dir, directory_digest=digest),
+    ))
+  result = output.dependencies[0]
+  console.print_stdout(f"Wrote file: {result.output_paths[0]}")
+  yield TestWorkspaceGoal(exit_code=0)
+
+
+class WorkspaceInConsoleRuleTest(ConsoleRuleTestBase):
```

Comment:
Neither of these tests actually test that the file exists... Could we add that assertion? Ideally in the first one by taking the path to create as a flag (so that the tempdir is controlled by the test, rather than deleted in the console rule, and we test the end-to-end) but if not, by having the second one check that the created files exist and have the correct contents?